hudi-cli/src/main/java/org/apache/hudi/cli/commands/MetadataCommand.java

```diff
@@ -0,0 +1,202 @@
+    HoodieTableMetaClient metaClient = HoodieCLI.getTableMetaClient();
+    Path metadataPath = new Path(HoodieMetadata.getMetadataTableBasePath(HoodieCLI.basePath));
+    try {
+      FileStatus[] statuses = HoodieCLI.fs.listStatus(metadataPath);
+    } catch (FileNotFoundException e) {
+      // Metadata directory does not exist
+      throw new RuntimeException("Metadata directory (" + metadataPath.toString() + ") does not exist.");
+    }
+
+    HoodieMetadata.remove(HoodieCLI.basePath);
+
+    long t1 = System.currentTimeMillis();
+    if (readOnly) {
+      HoodieMetadata.init(HoodieCLI.conf, HoodieCLI.basePath);
+    } else {
+      HoodieWriteConfig writeConfig = HoodieWriteConfig.newBuilder().withPath(HoodieCLI.basePath)
+          .withUseFileListingMetadata(true).build();
+      initJavaSparkContext();
+      HoodieMetadata.init(jsc, writeConfig);
+    }
+    long t2 = System.currentTimeMillis();
+
+    String action = readOnly ? "Opened" : "Initialized";
+    return String.format(action + " Metadata Table in %s (duration=%.2fsec)", metadataPath, (t2 - t1) / 1000.0);
+  }
+
+  @CliCommand(value = "metadata stats", help = "Print stats about the metadata")
+  public String stats() throws IOException {
+    HoodieTableMetaClient metaClient = HoodieCLI.getTableMetaClient();
+    Map<String, String> stats = HoodieMetadata.getStats(HoodieCLI.basePath, true);
+
+    StringBuffer out = new StringBuffer("\n");
+    out.append(String.format("Base path: %s\n", HoodieMetadata.getMetadataTableBasePath(HoodieCLI.basePath)));
+    for (Map.Entry<String, String> entry : stats.entrySet()) {
+      out.append(String.format("%s: %s\n", entry.getKey(), entry.getValue()));
+    }
+
+    return out.toString();
+  }
+
+  @CliCommand(value = "metadata list-partitions", help = "Print a list of all partitions from the metadata")
+  public String listPartitions() throws IOException {
+    HoodieTableMetaClient metaClient = HoodieCLI.getTableMetaClient();
+
+    StringBuffer out = new StringBuffer("\n");
+    if (!HoodieMetadata.exists(HoodieCLI.basePath)) {
+      out.append("=== Metadata Table not initilized. Using file listing to get list of partitions. ===\n\n");
+    }
+
+    long t1 = System.currentTimeMillis();
+    List<String> partitions = HoodieMetadata.getAllPartitionPaths(HoodieCLI.fs, HoodieCLI.basePath, false);
+    long t2 = System.currentTimeMillis();
+
+    int[] count = {0};
+    partitions.stream().sorted((p1, p2) -> p2.compareTo(p1)).forEach(p -> {
+      out.append(p);
+      if (++count[0] % 15 == 0) {
+        out.append("\n");
+      } else {
+        out.append(", ");
+      }
+    });
+
+    out.append(String.format("\n\n=== List of partitions retrieved in %.2fsec ===", (t2 - t1) / 1000.0));
+
+    return out.toString();
+  }
+
+  @CliCommand(value = "metadata list-files", help = "Print a list of all files in a partition from the metadata")
+  public String listFiles(
```

Comment:
@prashantwason : Can you make sure we can query when the underlying table is non-partitioned

Reply:
Yes. Add a unit test too.